### PR TITLE
Issue 471

### PR DESCRIPTION
**Pull request by lemire**
Fixes https://github.com/simdutf/simdutf/issues/471

### CODE REVIEW

**Comment by pauldreik** [comment]
this silences the test case, but the fuzzer immediately finds the same issue again so I do not think this is a sufficient fix.

**Comment by pauldreik** [comment]
this is a slightly modified testcase that triggers the code in this PR:
```cpp
TEST(issue_convert_utf8_to_utf32_8bad4f475a64f51e)
{
    const unsigned char data[] = {0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
                                  0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
                                  0xff, 0xff, 0xff, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
                                  0x20, 0x20, 0x20, 0x20, 0x20, 0xbb, 0x20, 0x9a, 0x9a, 0x9a, 0x9a,
                                  0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a,
                                  0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a,
                                  0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a,
                                  0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a,
                                  0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a,
                                  0x9a, 0x9a, 0x20, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x9a,
                                  0x9a, 0x9a, 0x9a, 0x9a, 0x9a, 0x8d, 0x20, 0xbb, 0x20, 0xbb, 0x20,
                                  0xbb, 0x20, 0x8d, 0x20, 0xb3, 0x20, 0xbb, 0x20};
    constexpr std::size_t data_len_bytes = sizeof(data);
    constexpr std::size_t data_len = data_len_bytes / sizeof(char);

    const auto validation1 = implementation.validate_utf8_with_errors((const char *) data, data_len);
    ASSERT_EQUAL(validation1.count, 22);
    ASSERT_EQUAL(validation1.error, simdutf::error_code::HEADER_BITS);

    const auto outlen = implementation.utf32_length_from_utf8((const char *) data, data_len);
    ASSERT_EQUAL(outlen, 47);
    std::vector<char32_t> output(outlen /* + 1*/);
    const auto r = implementation.convert_utf8_to_utf32((const char *) data,
                                                        data_len,
                                                        output.data());
    ASSERT_EQUAL(r, 0);
}

```

**Comment by lemire** [comment]
@pauldreik Unfortunate. My last PR will trade performance for extra safety. It should now be ok.

File: cmake/simdutf-flags.cmake [hunk 1]
```diff
@@ -8,8 +8,8 @@ if (NOT CMAKE_BUILD_TYPE)
   if(SIMDUTF_SANITIZE OR SIMDUTF_SANITIZE_UNDEFINED)
     set(CMAKE_BUILD_TYPE Debug CACHE STRING "Choose the type of build." FORCE)
     # SIMDUTF_SANITIZE only applies to gcc/clang:
-    message(STATUS "Setting debug optimization flag to -O1.")
-    set(CMAKE_CXX_FLAGS_DEBUG "-O1" CACHE STRING "" FORCE)
+    message(STATUS "Setting debug optimization flag to -O1 -g.")
+    set(CMAKE_CXX_FLAGS_DEBUG "-O1 -g" CACHE STRING "" FORCE)
```
Comment (Jarred-Sumner) [inline cmake/simdutf-flags.cmake:12]: fwiw I usually do -g3 in debug builds. `-g` with optimizations still misses stuff

**Pull request merged**